app/src/org/commcare/utils/FormUploadUtil.java

```diff
@@ -129,6 +129,7 @@ public static FormUploadResult sendInstance(int submissionNumber, File folder,
         if (files.length == 0) {
             Log.e(TAG, "no files to upload");
             listener.cancel(true);
+            throw new FileNotFoundException("Folder at path " + folder.getAbsolutePath() + " had no files.");
```

Comment:
@amstone326 throwing this exception results in the file being deleted here: https://github.com/dimagi/commcare-android/blob/20fbbe174fb9b9d47729665dec858de96ce37563/app/src/org/commcare/tasks/ProcessAndSendTask.java#L288-L288